Mathlib/Topology/MetricSpace/HoelderNorm.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2024 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import Mathlib.Topology.MetricSpace.Holder
+
+/-!
+# Hölder norm
+
+This file defines the Hölder (semi-)norm for Hölder functions alongside some basic properties.
+The `r`-Hölder norm of a function `f : X → Y` between two metric spaces is the least non-negative
+real number `C` for which `f` is `r`-Hölder continuous with constant `C`, i.e. it is the least `C`
+for which `WithHoelder C r f` is true.
+
+## Main definitions
+
+* `eHoelderNorm r f`: `r`-Hölder (semi-)norm in `ℝ≥0∞` of a function `f`.
+* `nnHoelderNorm r f`: `r`-Hölder (semi-)norm in `ℝ≥0` of a function `f`.
+* `MemHoelder r f`: Predicate for a function `f` being `r`-Hölder continuous.
+
+## Main results
+
+* `eHoelderNorm_eq_zero_iff`: the Hölder norm of a function is zero if and only if it is constant.
+* `MemHoelder.holderWith`: The Hölder norm of a Hölder function `f` is a Hölder constant of `f`.
+
+## Tags
+
+Hölder norm, Hoeder norm
+
+-/
+
+variable {X Y Z : Type*}
+
+open Filter Set
+
+open NNReal ENNReal Topology
+
+section PseudoEMetricSpace
+
+variable [PseudoEMetricSpace X] [PseudoEMetricSpace Y] {r : ℝ≥0} {f : X → Y}
+
+/-- The `r`-Hölder (semi-)norm in `ℝ≥0∞` of a function `f` is the least non-negative real
+number `C` for which `f` is `r`-Hölder continuous with constant `C`. This is `∞` if no such
+non-negative real exists. -/
+noncomputable
+def eHoelderNorm (r : ℝ≥0) (f : X → Y) : ℝ≥0∞ := ⨅ (C) (_ : HolderWith C r f), C
+
+/-- The `r`-Hölder (semi)norm in `ℝ≥0`. -/
+noncomputable
+def nnHoelderNorm (r : ℝ≥0) (f : X → Y) : ℝ≥0 := (eHoelderNorm r f).toNNReal
+
+/-- A function `f` is `MemHoelder r f` if it is Hölder continuous. Namely, `f` has a finite
+`r`-Hölder constant. This is equivalent to `f` having finite Hölder norm.
+c.f. `memHoelder_iff`. -/
+def MemHoelder (r : ℝ≥0) (f : X → Y) : Prop := ∃ C, HolderWith C r f
```

Comment:
Can you `MemHoelder.mono (hrs : r ≤ s) (hf : MemHoelder s f) : MemHoelder r f`?

Reply:
I do not want to prove this in this PR

Reply:
Oh right, it's not a simple application of `HolderWith.mono`